jena-arq/src/main/java/org/apache/jena/sparql/function/library/cdt/SizeFct.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jena.sparql.function.library.cdt;
+
+import org.apache.jena.cdt.CompositeDatatypeList;
+import org.apache.jena.cdt.CompositeDatatypeMap;
+import org.apache.jena.graph.Node;
+import org.apache.jena.sparql.expr.ExprEvalException;
+import org.apache.jena.sparql.expr.NodeValue;
+import org.apache.jena.sparql.function.FunctionBase1;
+
+public class SizeFct extends FunctionBase1
+{
+	@Override
+	public NodeValue exec( final NodeValue nv ) {
+		final Node n = nv.asNode();
+
+		final int size;
+		if ( CompositeDatatypeList.isListLiteral(n) ) {
+			size = CDTLiteralFunctionUtils.getList(n).size();
+		}
+		else if ( CompositeDatatypeMap.isMapLiteral(n) ) {
+			size = CDTLiteralFunctionUtils.getMap(n).size();
+		}
+		else {
+			throw new ExprEvalException("Neither a list nor a map literal: " + nv);
+		}
+
+		return NodeValue.makeInteger(size);
+	}
+
```

Comment:
Looks like this package/module has tabs instead of spaces, but I think that's OK.

Reply:
Fixed - thanks for catching that,

It's not critical and the codebase isnt perfect.

(IMO) github, and shared display of code generally, has tipped the balance in favour of spaces over tabs so everyone sees the code the same way.